test/Grpc.AspNetCore.Server.Tests/PipeExtensionsTests.cs

```diff
@@ -640,5 +662,57 @@ public Stream CreateDecompressionStream(Stream stream)
                 return _inner.CreateDecompressionStream(stream);
             }
         }
+
+        public class TestPipeReader : PipeReader
+        {
+            private readonly PipeReader _pipeReader;
+
+            public SequencePosition? Consumed { get; set; }
+            public SequencePosition? Examined { get; set; }
+
+            public TestPipeReader(PipeReader pipeReader)
+            {
+                _pipeReader = pipeReader;
+            }
+
+            public override void AdvanceTo(SequencePosition consumed)
+            {
+                Consumed = consumed;
+                Examined = null;
```

Comment:
This more asserts the calls made to the PipeReader than the behavior itself. That should be fine in this case though...

Reply:
Maybe it would be better just to set a bool that determines when AdvanceTo overload was called. Though I don't like this test, we shouldn't be storing sequence positions outside of AdvanceTo (they basically go away after this call).

Reply:
Updated to capture the underlying position index before AdvanceTo is called.